sqlserver/datadog_checks/sqlserver/statements.py

```diff
@@ -0,0 +1,311 @@
+import binascii
+import time
+import xml.etree.ElementTree as ET
+
+from cachetools import TTLCache
+
+from datadog_checks.base import is_affirmative
+from datadog_checks.base.utils.common import to_native_string
+from datadog_checks.base.utils.db.sql import compute_sql_signature
+from datadog_checks.base.utils.db.statement_metrics import StatementMetrics
+from datadog_checks.base.utils.db.utils import DBMAsyncJob, RateLimitingTTLCache, default_json_event_encoding
+from datadog_checks.base.utils.serialization import json
+
+try:
+    import datadog_agent
+except ImportError:
+    from ..stubs import datadog_agent
+
+DEFAULT_COLLECTION_INTERVAL = 10
```

Comment:
so we send events once every 10 seconds for sqlserver, but once a second for postgres/mysql? Is there a reason for the diff coll interval times or is just the convenience of having metrics/samples be collected in the same job?

Reply:
For postgres the 1s sample interval came from wanting to guarantee good coverage of a wide variety of queries. At 10 seconds we would miss out on shorter and less frequent queries. Whether we send the plans every second or every 10 seconds doesn't really make any difference. 

For sqlserver the plans are aggregated in the query plan cache (postgres doesn't have this) so we don't need to sample as frequently. Also, we only want to collect plans for queries that are actually happening the query/plan metrics tell us which plans actually ran since the last interval, which is why we use the result of the query metrics to feed into plan collection, and so it's all coupled together for SQL Server.

Reply:
ah okay, thanks that makes sense 👍